### PR TITLE
feat(providers): add mailtrap

### LINKED ISSUE #4325: 🚀 Feature: Add Mailtrap provider
### 🔖 Feature description

Mailtrap offers two services:
- email testing
- email sending (for production emails)

I would love to see provider for their email sending service.

### 🎤 Why is this feature needed ?

Mailtrap is quite well known company in email sending space.

### ✌️ How do you aim to achieve this?

I will implement Mailtrap provider for email channel [just like any other provider](https://docs.novu.co/community/add-a-new-provider#how-to-add-a-new-provider).

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by michaldziuba03**
### What change does this PR introduce?

This PR introduces new provider for email channel - Mailtrap.

They offer two services:
- email testing
- email sending (for production emails)

This PR creates provider for their [Email Sending service](https://mailtrap.io/email-sending/).

API reference: https://api-docs.mailtrap.io/docs/mailtrap-api-docs/67f1d70aeb62c-send-email
Mailtrap SDK: https://github.com/railsware/mailtrap-nodejs

### Why was this change needed?

Closes #4325

#### Workflow with provider

Workflow

#### Integration store with `IS_MULTI_PROVIDER_CONFIGURATION_ENABLED=true`

integration-store1

#### Integration store default view

integration-store2

#### Email sent with attachement

email-screenshot

### CODE REVIEW

File: providers/mailtrap/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/** @type {import('ts-jest/dist/types').InitialOptionsTsJest} */
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+  moduleNameMapper: {
+    axios: 'axios/dist/node/axios.cjs',
```
Comment (michaldziuba03) [inline providers/mailtrap/jest.config.js:6]: without that option Jest has problem with importing `axios` inside [Mailtrap SDK](https://github.com/railsware/mailtrap-nodejs).

File: providers/mailtrap/package.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
+{
+  "name": "@novu/mailtrap",
+  "version": "0.20.0-alpha.0",
```
Comment (michaldziuba03) [inline providers/mailtrap/package.json:3]: provider template generator is "outdated", so I changed to that version manually.

File: providers/mailtrap/src/lib/mailtrap.provider.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  async sendMessage(
+    options: IEmailOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const response = await this.sendWithMailtrap(options);
+
+    return {
+      id: response.message_ids[0],
+      date: new Date().toISOString(),
```
Comment (michaldziuba03) [inline providers/mailtrap/src/lib/mailtrap.provider.ts:54]: mailtrap doesn't return own timestamp

**Comment by scopsy** [comment]
That's a fantastic idea @michaldziuba03! I will let the rest of the team to review this, but amazing work

**Comment by michaldziuba03** [comment]
@BiswaViraj PR for docs is ready https://github.com/novuhq/docs/pull/157

**Comment by BiswaViraj** [comment]
@michaldziuba03 Amazing,  Can you please fix the conflicts So that I can merge?

**Comment by michaldziuba03** [comment]
@BiswaViraj I fixed that conflict, I hope that no git conflicts arise in the meantime :)

**Pull request merged**